modules/rhsm_manifest.py

```diff
@@ -0,0 +1,271 @@
+           - Manifest Name
+    uuid:
+        description:
+           - Manifest uuid
+    username:
+        description:
+            - Username on Foreman server
+        required: true
+    password:
+        description:
+            - Password for user accessing Foreman server
+        required: true
+    pool_id:
+        description:
+            - Subscription pool_id
+    quantity:
+        description:
+            - quantity of pool_id Subscriptions
+        type: int
+    state:
+        description:
+            - State of the Subscription
+        default: present
+        choices:
+            - present
+            - absent
+    path:
+        description:
+            - path to export the manifest
+        type: path
+    validate_certs:
+        description:
+            - Validate Portal Certs
+        default: True
+        type: bool
+    portal:
+        description:
+            - Red Hat Portal subscription access address
+        default: https://subscription.rhn.redhat.com
+'''
+
+EXAMPLES = '''
+- name: Create katello.example.com Manifest and add 7 sub
+  rhsm_manifest:
+    name: "katello.example.com"
+    username: "john-smith"
+    password: "changeme"
+    pool_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+    quantity: 7
+    state: present
+    validate_certs: false
+
+- name: Ensure my manifest has 10 of one subs in it and export
+  rhsm_manifest:
+    uuid: XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
+    username: john-smith
+    password: changeme
+    pool_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+    quantity: 10
+    state: present
+    validate_certs: false
+    path: /root/manifest.zip
+
+- name: Remove all of one subs from katello.example.com
+  rhsm_manifest:
+    name: katello.example.com
+    username: john-smith
+    password: changeme
+    pool_id: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+    state: absent
+    validate_certs: false
+'''
+
+RETURN = '''# '''
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import fetch_url
+from ansible.module_utils.pycompat24 import get_exception
+
+
+def fetch_portal(module, path, method, data={}, accept_header='application/json'):
+    url = module.params['portal'] + path
+    headers = {'accept': accept_header,
+               'content-type': 'application/json'}
+    resp, info = fetch_url(module, url, json.dumps(data), headers, method)
+    if resp is None:
+        try:
+            error = json.loads(info['body'])['displayMessage']
+        except:
+            error = info['msg']
+        raise Exception(error)
+    return resp, info
+
+
+def create_manifest(module):
+    path = "/subscription/consumers"
+    data = {'name': module.params['name'],
+            'type': "satellite",
+            'owner': module.params['rhsm_owner'],
+            'facts': {'distributor_version': 'sat-6.0',
+                      'system.certificate_version': '3.2'}}
```

Comment:
Should either of these be configurable to protect against future changes in the portal?

Reply:
Maybe..  

I have no idea what `system.certificate_version` actually does?
`distributor_version` should probably be, though I'm not sure what API we need to call to modify it on an already existing manifest, added a TODO comment for now.

Most of the APIs I got form https://github.com/Katello/hammer-cli-csv/commit/648163df7fe0dc0ead3ea0186eaba6d10d7f7b96. Maybe @thomasmckay has some insights?